Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -183,6 +194,18 @@ namespace AZ
             m_isDirty = false;
         } 
 
+        void MaterialAsset::ApplyVersionUpdates()
+        {
+            for (int i = 0; i < aznumeric_cast<int>(m_materialTypeAsset->GetVersion() - m_materialTypeVersion); ++i)
+            {
+                const auto& versionUpdate = m_materialTypeAsset->GetMaterialVersionUpdate(m_materialTypeVersion + i + 1);
+
+                versionUpdate.ApplyVersionUpdates(*this);
+            }
```

Comment:
Suggestion: this would be cleaner...
```suggestion
            for (int i = m_materialTypeVersion; i < m_materialTypeAsset->GetVersion(); ++i)
            {
                const auto& versionUpdate = m_materialTypeAsset->GetMaterialVersionUpdate(i + 1);

                versionUpdate.ApplyVersionUpdates(*this);
            }
```